src/SDKs/SqlManagement/Sql.Tests/Utilities/TestEnvironmentUtilities.cs

```diff
@@ -9,6 +9,75 @@ namespace Sql.Tests
 {
     public static class TestEnvironmentUtilities
     {
+        private const string _environmentVariableName = "TEST_CSM_ORGID_AUTHENTICATION";
+
+        private static readonly TestEnvironment _environment = 
+            new TestEnvironment(Environment.GetEnvironmentVariable(_environmentVariableName));
+
+        // We now load default locations from environment variable.
+        // Tests were recorded with this appended to DefaultLocationId=japaneast;DefaultLocation=Japan East;DefaultSecondaryLocationId=centralus;DefaultSecondaryLocation=Central US;DefaultStagePrimaryLocation=North Europe;DefaultStageSecondaryLocation=SouthEast Asia;DefaultEuapPrimaryLocation=East US 2 EUAP;DefaultEuapPrimaryLocationId=eastus2euap
+        public static string DefaultLocation
+        {
+            get
+            {
+                return GetValueFromEnvironment("DefaultLocation");
```

Comment:
- Yes, they will need to know.  If they try to run the tests in record without these values set, they will get a very clear message that the required value is missing from the environment variable, and they will be told the name of the environment variable that needs to have the value.

- I still don't understand why we can't hard-code location in the test utilities.  It doesn't affect the usability of the test suite.  It's very easy to tell what location is being used for tests from reading the code/recordings.  However, because this was a request from the SDK team, I made this change.

- When we last discussed this, you outlined this exact approach, which is why I did it this way.  I have refreshed that email in your inbox.